node/external/blockAPI/metaBlock.go

```diff
@@ -102,15 +103,15 @@ func (mbp *metaAPIBlockProcessor) convertMetaBlockBytesToAPIBlock(hash []byte, b
 			SourceShard:      mb.SenderShardID,
 			DestinationShard: mb.ReceiverShardID,
 		}
-		if withTxs {
+		if options.WithTransactions {
 			miniBlockCopy := mb
-			mbp.getAndAttachTxsToMb(&miniBlockCopy, headerEpoch, miniblockAPI)
+			mbp.getAndAttachTxsToMb(&miniBlockCopy, headerEpoch, miniblockAPI, options)
```

Comment:
Something like this (but without the extractions to `miniblocks.go`)?

https://github.com/ElrondNetwork/elrond-go/pull/4145

Looks a bit off. We can refactor the loading of miniblocks from both units (miniblocks unit and receipts unit), plus the loading of transactions into separate objects such as `miniblocksRepository` and `transactionsRepositoy`, then handle the `with*` flags at a higher level, in `blockAPI/shardBlock.go` and `blockAPI/metablock.go` (e.g. which somehow resemble the layer between classic web controllers and the data access layer). However, this refactoring would not be trivial, especially because it involves movement of the code that deals with `batch.Batch{}` - and moving it now also means fixing it now - a bit out of the scope of this PR.

What do you feel about this? Merge the linked PR into this one or postone the refactoring in `blockAPI`?

Reply:
Okay, we can close the #4145 PR and this feat request can be safely postponed :+1: